File: src/main/java/team4/footwithme/resevation/service/response/MercenaryResponse.java
package team4.footwithme.resevation.service.response;

import team4.footwithme.resevation.domain.Mercenary;

public record MercenaryResponse(
        Long mercenaryId,
        Long reservationId,
        String description) {
    public MercenaryResponse(Mercenary mercenary) {
        this(
                mercenary.getMercenaryId(),
                mercenary.getReservation().getReservationId(),
                mercenary.getDescription()
        );
    }
}

File: src/main/java/team4/footwithme/resevation/service/response/MercenaryResponse.kt
package team4.footwithme.resevation.service.response

import team4.footwithme.resevation.domain.Mercenary

@JvmRecord
data class MercenaryResponse(
    val mercenaryId: Long?,
    val reservationId: Long?,
    val description: String?
) {
    constructor(mercenary: Mercenary?) : this(
        mercenary!!.mercenaryId,
        mercenary.reservation!!.reservationId,
        mercenary.description
    )

}
